src/mono/wasi/Wasi.Build.Tests/WasiTemplateTests.cs

```diff
@@ -50,7 +50,7 @@ public void ConsoleBuildAndRunAOT(string config, bool singleFileBundle)
                         CreateProject: false,
                         Publish: false,
                         TargetFramework: BuildTestBase.DefaultTargetFramework));
-        RunWithoutBuild(config, id);
```

Comment:
We need HTTP here, because we didn't solve the trimming yet, right?

Reply:
This is build without native and without publish, and so it doesn't trim/re-link.
The default runtime binary has all the dependency.
This is how "demo mode" experience would look like going forward, unless WASI brings optional imports.

Reply:
Alternatively we could ship multiple flavors of the binaries. But that leads to exponential explosion on feature flags
- [ ] http client
- [ ] socket
- [ ] threads
- [ ] invariant globalization
- [ ] timezones
- [ ] wasi:clock - timers
- [ ] going forward probably also file system will become separate from `wasi:cli`
- [ ] ...

Or we can say that the "demo mode" is just `wasi:cli` - that is console app, nothing else. If computation is all you need.